src/resolve-info.ts

```diff
@@ -37,19 +39,32 @@ export interface ResolveInfoEnricherInput {
   parentType: GraphQLObjectType;
   returnType: GraphQLOutputType;
   fieldName: string;
-  fieldNodes: FieldNode[];
+  fieldNodes: JitFieldNode[];
 }
 
-export interface FieldExpansion {
+export type ShouldIncludeVariables = {
+  variables: CoercedVariableValues;
+};
+
+export interface ShouldIncludeExtension {
+  __shouldInclude: (variables: ShouldIncludeVariables) => boolean;
+}
+
+export type FieldExpansion = ShouldIncludeExtension & {
   // The possible return types that the field can return
   // It includes all the types in the Schema that intersect with the actual return type
   // eslint-disable-next-line no-use-before-define
   [returnType: string]: TypeExpansion;
-}
+};
+
+type RootFieldExpansion = {
```

Comment:
Why do we need this separate type?

Reply:
initially it was because root doesn't have ShouldIncludeExtension since root is always included.